internal/graph/check.go

```diff
@@ -371,9 +378,9 @@ func (c *LocalChecker) ResolveCheck(
 func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckRequest) CheckHandlerFunc {
 
 	return func(ctx context.Context) (*openfgapb.CheckResponse, error) {
-		typesys, ok := typesystem.TypesystemFromContext(parentctx) // note: use of 'parentctx' not 'ctx' - this is important
-		if !ok {
-			return nil, fmt.Errorf("typesystem missing in context")
+		typesys, err := c.typesysResolver(ctx, req.GetStoreID(), req.GetAuthorizationModelID())
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve typesystem: %v", err)
```

Comment:
It's part of the `typesystem` package though, and that's an internal bit. The server (and by extension commands/queries) layer should be responsible for user facing error messages, since those are the code bits that handle the public API itself.

Reply:
It's the same reason why the `internal/graph` package doesn't return user facing errors (e.g. errors from the `pkg/server/errors` package).